harpy/bin/bx_to_end.py

```
@@ -62,20 +37,45 @@ def format_fastq(record):
+
+    if is_fastq:
+        with (
+            pysam.FastxFile(infile, persist=False) as fq_in,
+            gzip.GzipFile(fileobj= sys.stdout.buffer, mode= "wb", compresslevel=6) as fq_out
+        ):
+            for rec in fq_in:
+                fq_out.write(format_fastq(rec).encode())
+    else:
+        try:
+            bam_in = pysam.AlignmentFile(infile, "rb")
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)